src/components/loginPage/LoginPage.jsx

```diff
@@ -1,5 +1,78 @@
+import { useRef } from "react";
+import { Link } from "react-router-dom";
+import { ButtonWithLoader } from "../buttons/Buttons";
+
 import "./css/loginPageStyle.css";
+import LoginLogic from "./logic/loginLogic";
 
 export default function LoginPage() {
-  return <h2>Login</h2>;
+  const emailInputRef = useRef();
+  const passwordInputRef = useRef();
+  const { login, loggingIn, serverResponse } = LoginLogic();
+  return (
+    <main className="main --login-signup --verticle-flex --centered-flex">
+      <div className="login-signup-outer-container">
+        <div className="login-box-container --horizontal-flex">
+          <section className="section --login-with --verticle-flex --centered-flex --has-gap">
+            <button className="btn --with-google --primary-btn --has-hover-overlay">
+              Continue with Google
+            </button>
+            <button className="btn --with-apple --primary-btn --has-hover-overlay">
+              Continue with Apple
+            </button>
+            <button className="btn --with-facebook --primary-btn --has-hover-overlay">
+              Continue with Facebook
+            </button>
+          </section>
+          <section className="section --login-box --verticle-flex --centered-flex --has-gap">
+            <h3 className="sub-heading --h3">Login</h3>
+            <div className="--verticle-flex --has-gap">
+              <input
+                className="input"
+                type="text"
+                name="email"
+                placeholder="Email"
+                ref={emailInputRef}
+              />
+              <input
+                className="input"
+                type="password"
+                name="password"
+                id="password"
+                placeholder="Password"
+                ref={passwordInputRef}
```

Comment:
What exactly is ref over here?

Reply:
It's a reference to that input element. I had to use it like this for the animations. Couldn't think of another approach.